### DISCUSSION TITLE
No support for litestream (sqlite streaming replication)

### DISCUSSION THREAD

**Discussion by James-Quigley** [dani-garcia/vaultwarden#1519] [Ideas]
### Subject of the issue

Attempting to use [litestream](https://litestream.io/) corrupts the sqlite database. It would be very useful to have this support to make for really easy database backups. Litestream has a [tips/tricks page](https://litestream.io/tips/) for getting your application to support litestream

### Deployment environment

* bitwarden_rs version: 1.19.0

* Install method: Docker image

* Clients used: 

* Reverse proxy and version: 

* MySQL/MariaDB or PostgreSQL version: 

* Other relevant details:

### Steps to reproduce

Run bitwarden_rs, run litestream replicate

### Expected behaviour

No database corruption

### Actual behaviour

Database corruption

**Comment by BlackDex** [comment]
This is more a feature request then an issue.

To have this supported we probably need to add a new configuration parameter or something special to the database url.
So, something like after the database file `DATABASE_URL=/data/db.sqlite3;busy_timeout=5000,synchronous=NORMAL` could be an option.

That way people can also change more PRAGMA settings if they really want.

**Comment by jjlin** [comment]
I would go further -- this is all feature request, zero issue.

I'm not sure what exactly is meant by "database corruption", but https://litestream.io/tips/ only seems to say that WAL mode is required, which bitwarden_rs already uses by default. The other settings seem to be for better performance (the `PRAGMA synchronous` setting) or better robustness in the main application (the `PRAGMA busy_timeout` setting). So this could well be a bug in Litestream instead. (I personally wouldn't entrust my backup regimen primarily to a program that's only been released for 3 months, but if you want to, it's your funeral.)

If bitwarden_rs were to provide more user control over database init, it would probably be better to provide a new config item like `DATABASE_INIT_FILE` that points to a file containing SQL statements to run on startup, then use `diesel::sql_query` to run each non-blank/comment line. This would provide people with yet another way to shoot themselves in the foot, though.

**Comment by sl33nyc** [comment]
Has anyone tried litestream with vaultwarden in the past 2+ years?  If so, can you comment about your experience?
